server/remote_cache/byte_stream_server/byte_stream_server.go

```diff
@@ -279,7 +278,7 @@ func (s *ByteStreamServer) initStreamState(ctx context.Context, req *bspb.WriteR
 	}
 
 	var committedWriteCloser interfaces.CommittedWriteCloser
-	if r.GetDigest().GetHash() != digest.EmptySha256 && !exists {
+	if !digest.IsEmpty(r.GetDigest()) && !exists {
```

Comment:
```suggestion
	if !digest.IsEmpty(r.GetDigest()) {
```

Nit: simplify. If exist, the function would have returned in the if clause above

Reply:
then, it's probably easier to read if you flip the if/else so that we could remove the negation `!`